Bullwinkle.class.nut

```diff
@@ -385,7 +395,7 @@ class Bullwinkle {
 
             // Add the retry information
             message.retry <- {
-                "ts": time() + timeout,
+                "ts": ( Bullwinkle._isAgent() ? time() : hardware.micros()/1000000 )  + timeout,
```

Comment:
I guess I would argue that the semantics of the message.ts are rather useless as is and this pull request makes them useful.  It seems silly to send a useless message.ts with each payload AND include a custom message.data.ts with the timestamp the application cares about.

If the message.ts does not correspond to the time that the data was generated, why send it over the wire at all?  Bullwinkle.Package._ts is what all the underlying logic is using (in this pull request).